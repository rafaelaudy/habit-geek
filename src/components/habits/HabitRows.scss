@import "../Variables.scss";

.habit-row {
  display: flex;

  &:first-child .habit-row-frequency__cell--today {
    border-top-width: $border-width;
    border-top-left-radius: 0.25rem;
    border-top-right-radius: 0.25rem;
  }

  &:last-child .habit-row-frequency__cell--today {
    border-bottom-width: $border-width;
    border-bottom-left-radius: 0.25rem;
    border-bottom-right-radius: 0.25rem;
  }

  &__title {
    flex-grow: 1;
    display: flex;
    align-items: center;
  }
}

@media (max-width: 576px) {
  .habit-row {
    flex-wrap: wrap;
    margin: auto;
    max-width: 380px;

    &__title {
      justify-content: center;
      height: 42px;
    }
  }
}
