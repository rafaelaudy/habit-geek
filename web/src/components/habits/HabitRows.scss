@import "../Variables.scss";

.habit-row {
  &__header,
  & {
    display: flex;

    &:last-child .habit-row-frequency__cell--today {
      border-bottom-width: $border-width;
      border-bottom-left-radius: 0.25rem;
      border-bottom-right-radius: 0.25rem;
    }
  }

  &__title {
    flex-grow: 1;
    display: flex;
    align-items: center;
  }
}

@media (max-width: 576px) {
  .habit-row {
    &__header,
    & {
      flex-wrap: wrap;
      margin: auto;
      max-width: 380px;
    }

    &__header .habit-row__title {
      height: 0;
    }

    &__title {
      justify-content: center;
      height: 42px;
    }
  }
}
