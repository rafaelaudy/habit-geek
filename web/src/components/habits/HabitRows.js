import React from "react";

import HabitRowActions from "./HabitRowActions";
import HabitRowFrequency from "./HabitRowFrequency";
import "./HabitRows.scss";

const HabitRows = ({
  week,
  habits,
  isReadOnly,
  toggleDayHabit,
  onUpdateHabit
}) =>
  habits.map(
    (
      { name, frequency, checked, habitSucceded, habitFailed },
      containerIndex
    ) => {
      return (
        <div key={`check-container-${containerIndex}`} className="habit-row">
          <div className="habit-row__title">
            <span>
              ({frequency}x) - {name}
            </span>
          </div>

          <HabitRowFrequency
            week={week}
            name={name}
            checked={checked}
            habitSucceded={habitSucceded}
            habitFailed={habitFailed}
            isReadOnly={isReadOnly}
            toggleDayHabit={toggleDayHabit}
          ></HabitRowFrequency>

          {isReadOnly ? null : (
            <HabitRowActions
              week={week}
              name={name}
              onUpdateHabit={onUpdateHabit}
            ></HabitRowActions>
          )}
        </div>
      );
    }
  );

export default HabitRows;
