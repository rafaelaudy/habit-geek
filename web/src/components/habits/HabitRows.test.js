import React from "react";
import { shallow } from "enzyme";
import HabitRows from "./HabitRows";

describe("HabitRows component", () => {
  it("Renders static elements", () => {
    const habitRows = shallow(
      <HabitRows
        week="week"
        habits={[
          {
            name: "name",
            frequency: "frequency",
            checked: "checked",
            habitSucceded: "habitSucceded",
            habitFailed: "habitFailed"
          }
        ]}
        isReadOnly={false}
        toggleDayHabit={() => {}}
        onUpdateHabit={() => {}}
      />
    );
    expect(habitRows).toMatchSnapshot();
  });

  it("Hides action container when is in readonly mode and disables checkboxes", () => {
    const habitRows = shallow(<HabitRows habits={[{}]} isReadOnly={true} />);
    expect(habitRows.find("HabitRowActions").length).toBe(0);
  });

  it("Shows the correct ammount of habits", () => {
    const checked = [false, false, false, false, false, false, false];
    const habitRows = shallow(
      <HabitRows
        habits={[
          { name: "exercise", frequency: 1, checked },
          { name: "read", frequency: 1, checked }
        ]}
      />
    );
    expect(habitRows.find(".habit-row").length).toBe(2);
  });
});
