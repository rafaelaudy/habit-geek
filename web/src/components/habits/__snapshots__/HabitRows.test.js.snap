// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitRows component Renders static elements 1`] = `
<div
  className="habit-row"
  key="check-container-0"
>
  <div
    className="habit-row__title"
  >
    <span>
      (
      frequency
      x) - 
      name
    </span>
  </div>
  <HabitRowFrequency
    checked="checked"
    habitFailed="habitFailed"
    habitSucceded="habitSucceded"
    isReadOnly={false}
    name="name"
    toggleDayHabit={[Function]}
    week="week"
  />
  <HabitRowActions
    name="name"
    onUpdateHabit={[Function]}
    week="week"
  />
</div>
`;
